inference_levels/local_inferencers/local_inference.py

```diff
@@ -0,0 +1,35 @@
+import torch
+
+
+class BaseLocalInference(torch.nn.Module):
+    def __init__(self,embedding_size):
+        super().__init__()
+
+        self.energy_metrix = None
+
+
+        self.zero_dim_softmax = torch.nn.Softmax(0)
```

Comment:
these fields can also be removed.
you can use torch.nn.functional.softmax instead, as a function -- there is not reason to have this as a separate object (it only increases cognitive, since reading 'torch.nn.functional.softmax(stuff) does not require looking at class initialization to understand where self.zero_dim_softmax comes from)